### DISCUSSION TITLE
validateForm  accept a different validateSchema

### DISCUSSION THREAD

**Discussion by matheusleandroo** [jaredpalmer/formik#2753] [General]
Hi!

I think it would be a good idea if the validateForm function could receive a different validationSchema as a parameter (otherwise it takes the default).

**Comment by harshagarwalnyu** [comment]
`validateForm()` doesn't accept a different `validationSchema` — it always uses the one on the `<Formik>` component. But you can run validation against a different schema manually using Formik's built-in helpers:

```tsx
import { useFormikContext } from 'formik';
import { validateYupSchema, yupToFormErrors } from 'formik';
import * as Yup from 'yup';

function MyComponent() {
  const { values, setErrors } = useFormikContext();

  async function validateWithAlternateSchema() {
    const alternateSchema = Yup.object({
      email: Yup.string().email().required(),
      // ... different rules
    });

    try {
      await validateYupSchema(values, alternateSchema, true);
      setErrors({});  // no errors
      return true;
    } catch (err) {
      const errors = yupToFormErrors(err);
      setErrors(errors);
      return false;
    }
  }

  return (
    <button type="button" onClick={validateWithAlternateSchema}>
      Validate with alternate schema
    </button>
  );
}
```

`validateYupSchema` and `yupToFormErrors` are exported from `formik` — they're the same functions Formik uses internally.

**If you use Zod instead of Yup:**
```ts
import { toFormikValidationSchema } from 'zod-formik-adapter';

const alternateSchema = z.object({ ... });
const errors = await alternateSchema.safeParseAsync(values);
if (!errors.success) {
  const formikErrors = {};
  errors.error.issues.forEach(i => {
    set(formikErrors, i.path.join('.'), i.message);
  });
  setErrors(formikErrors);
}
```

**Feature request note:** this pattern is intentional — if you find yourself needing different schemas for different actions, consider splitting into multiple `<Formik>` instances or using `validate` prop (function) instead of `validationSchema`, since functions are easier to compose.
